pkg/config/configuration.go

```diff
@@ -156,18 +156,18 @@ const (
 	APIAccessRuleProtocolGRPC APIAccessRuleProtocol = "grpc"
 )
 
-// GetRulesByProtocol returns a list of APIAccessRule objects filtered by protocol
-func (r APIAccessRules) GetRulesByProtocol(protocol APIAccessRuleProtocol) []APIAccessRule {
-	res := make([]APIAccessRule, len(r))
-	n := 0
+// GetRulesByProtocol returns a list of APIAccessRule objects for a protocol
+// The result is a map where the key is in the format "<version>/<endpoint>"
+func (r APIAccessRules) GetRulesByProtocol(protocol APIAccessRuleProtocol) map[string]struct{} {
+	res := make(map[string]struct{}, len(r))
 	for _, v := range r {
 		//nolint:gocritic
 		if strings.ToLower(string(v.Protocol)) == string(protocol) {
-			res[n] = v
-			n++
+			key := v.Version + "/" + v.Name
+			res[key] = struct{}{}
```

Comment:
please specifically add unit tests for these changes.

Reply:
Done